2149: Do not redirect on list action if disabled for default backend

laurent-bientz opened the pull request:
#2149, this PR avoid to redirect on list action of the default entity if disabled.

Now it get the params defined in the YAML (eg: ```params: { action: 'edit', id: 1 }```) to override the target action and the id if defined.

@javiereguiluz I'm not rly confident about the way I implement it because when you get the first Entity (```array_keys($backendConfig['entities'])[0]```), we don't have the params defined at this level.

So I've to create a new recursive function (```processDefaultEntityParams```) which loop on the design menu configuration to find the matching entity and to check if it have params defined.

I added two additional settings so now we have:
* $backendConfig['default_entity_name'] > your entity
* $backendConfig['default_entity_action'] > the entity action (list if ok, else the one defines in params)
* $backendConfig['default_entity_id'] > the entity id (if edit)

Then, in ```processDefaultHomepage```, I put the correct action and the correct id if needed.

Tell me what u think.

Cheers

Pierstoval commented:
Can you add more details in the docs & possibly add tests for this feature so it's better documented and understandable besides source code? 😄

laurent-bientz commented:
@Pierstoval sry didn't see your message.

What kind of docs do you want I add? because I added no new configurations at all.

I'm just looping on the design menu configuration to find the first entity and to check if it have params defined to forward on the correct wanted action.

eg:
```yaml
easy_admin:
    design:
        menu:
            - { label: 'Page', entity: app_careerspage, params: { action: 'edit', id: 1 } }
            - #...
```

Before these parameters produced an error, cause we were trying to forward on `List` action instead on `Edit` (id: 1).

That the point I fixed in this PR.

For adding tests, sorry I'm not very comfortable with that.

Pierstoval commented on src/Configuration/DefaultConfigPass.php:
```diff
@@ -41,9 +41,43 @@ private function processDefaultEntity(array $backendConfig)
+     *
+     * @return array
+     */
+    private function processDefaultEntityParams(array $items, $firstEntityName)
```
I'd rename `$items` to `$menu`, as it's precisely the name of the config option you retrieve.

Pierstoval commented:
Ok I had not checked the linked issue, then for me it's ok, even with my comment :smile:

shakaran commented:
@laurent-bientz could you update the branch for avoid the merge conflicts?

javiereguiluz commented:
I'm sorry we didn't merge this pull request. Now it's too late, so let's close it.
